server/src/main/java/org/opensearch/index/codec/CodecAliases.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.codec;
+
+import org.apache.lucene.codecs.Codec;
+
+import java.util.Set;
+
+/**
+ * This {@link CodecAliases} to provide aliases for the {@link Codec}.
+ *
+ * @opensearch.internal
+ */
+public interface CodecAliases {
```

Comment:
Should this be `@ExperimentalApi` or `@PublicApi`?

@reta what do you think?

Reply:
@ExperimentalApi for sure, thanks @andrross

Reply:
@andrross / @reta - Curious why this should be experimental?

Reply:
@backslasht this is a new API (and we do plan some more work on adding more plugins), it is safer to keep door open for change for the time being, with `@ExperimentalApi` allows that.

Reply:
@backslasht The word experimental here is definitely overloaded. In this context it doesn't say anything about the user-facing features that depend on this API (zstd compression in this case); it is instead saying this brand new Java API related to defining codecs may change in backward incompatible ways so plugin implementers need to be prepared to absorb breaking changes at any time until this matures.

Reply:
Got it. Thanks for the explanation @andrross